vpd-tool/include/tool_utils.hpp

```diff
@@ -125,5 +125,63 @@ inline std::string getPrintableValue(const types::BinaryVector& i_keywordValue)
 
     return l_oss.str();
 }
+
+/**
+ * @brief API to read keyword's value from hardware.
+ *
+ * This API reads keyword's value by requesting DBus service(vpd-manager) who
+ * hosts the 'ReadKeyword' interface to read keyword's value.
+ *
+ * @param[in] i_serviceName - Name of the Dbus service hosting 'ReadKeyword'
+ * method.
+ * @param[in] i_objectPath - Object path under the service.
+ * @param[in] i_interface - Interface under which 'ReadKeyword' method exist.
+ * @param[in] i_eepromPath - EEPROM file path.
+ * @param[in] i_paramsToReadData - Property whose value has to be read.
+ *
+ * @return - Value read from Dbus
+ *
+ * @return std::runtime_error
+ */
+inline types::DbusVariantType readKeywordFromHardware(
+    const std::string& i_serviceName, const std::string& i_objectPath,
+    const std::string& i_interface, const std::string& i_eepromPath,
+    const types::ReadVpdParams i_paramsToReadData)
+{
+    // Mandatory fields to make a dbus call.
+    if (i_serviceName.empty() || i_objectPath.empty() || i_interface.empty() ||
```

Comment:
Is it required to pass these parameters? These will be fixed as the value is always to be read from VPD manager, at least all these checks will se saved and not required.

Reply:
updated